docs/design/design-shifuFSM.md

```diff
@@ -0,0 +1,165 @@
+# Shifu FSM Design
+Shifu FSM stands for Shifu Finite State Machine. Shifu FSM stands for Shifu Finite State Machine. It is a workflow control system that allows users to control the behavior of the device based on different states.
+
+## Control Flow
+```mermaid
+flowchart LR
+stateA -->|action1| stateB
+stateA -->|action2| stateC
+stateC -->|action3| stateA
+
+```
+
+### State
+A state is a condition or status of the system. It represents a specific moment or situation in the execution of the machine.
+
+### Action
+An action here is just an instruction in the configmap. Each state can define a set of instructions along with the next state the device will end up in when receiving the action. If a state receives an unsupported action, it will simply rejects the action.
+
+Take the above control flow as an example, `stateA` receives `action1` and transit to `stateB`. `stateA` receives `action2` and transit to `stateC`. `stateC` receives `action3` and transit back to `stateA`.
+
+
+## High level design
+We can reuse a lot of existing design and implementations to achieve the FSM functionality. We can embed the entire FSM as part of our configmap, along with instructions and telemetries. 
+We can add a state field inside `edgedeivce` to represent current state of the device, and since edgedevice is stored in `etcd`, it's also good for fail-recovery.
+
+## Detailed Design
+### Configmap
```

Comment:
also put an example of configmap yaml path in shifu repository here?

Reply:
The configmap here is already a valid yaml file, don't think it is necessary to put another example.